designs/2024-hooks-for-test-cases/README.md

```diff
@@ -0,0 +1,311 @@
+- Repo: eslint/eslint
+- Start Date: 2024-08-12
+- RFC PR: https://github.com/eslint/rfcs/pull/122
+- Authors: [Anna Bocharova](https://github.com/RobinTail)
+
+# Hooks for test cases
+
+## Summary
+
+<!-- One-paragraph explanation of the feature. -->
+
+I propose adding an optional `setup` property to the test cases that the ESLint `RuleTester` runs. This feature
```

Comment:
Curious about naming: why `setup` vs. `before`? 

I have a slight preference for `before` and `after`, as I think these more clearly indicate what's happening. We also don't need to worry about capitaliziation issues `setUp`, `tearDown`.

Reply:
I don't think grammar rules are a deciding factor in this decision. What I care about is what will be most meaningful and obvious to the majority of users. 

It's fairly common among older tests runners to use `setUp` and `tearDown`, whereas newer ones tend to use `before` and `after` (or `beforeEach`, `beforeAll`, etc.). That's partly why I think `before` and `after` may be a better choice for this.

Reply:
OK.

Reply:
addressed in f7ac4af